demes/src/specification.rs

```diff
@@ -1059,6 +1093,26 @@ impl Deme {
     pub fn proportions(&self) -> &[Proportion] {
         &self.proportions
     }
+
+    /// Size of Deme at a given time
+    pub fn size_at<F: Into<f64>>(&self, time: F) -> Result<Option<DemeSize>, DemesError> {
+        let time: f64 = time.into();
+        Time::from(time).validate(DemesError::DemeError)?;
+
+        if time == f64::INFINITY && self.start_time == f64::INFINITY {
+            return Ok(Some(self.epochs()[0].start_size));
+        };
+
+        let epoch = self.epochs().iter().find(|x| {
+            x.time_interval()
+                .contains_exclusive_start_inclusive_end(time)
+        });
+
+        match epoch {
+            None => Ok(None),
+            Some(e) => Ok(Some(e.size_at(f64::from(time))?)),
```

Comment:
ah that's weird, I do have rust_analyzer set up and clippy that runs on save. It usually does a lot of reformatting on save so I thought it worked but does not flag this.

Reply:
These lints are treated as warnings when you are working locally.  The CI runs clippy in a "warnings are errors" mode.  In my editor (neovim), clippy warnings are indicated visually in the error and populate the diagnostics list so that I can search the project and rapidly go to each warning/error.